File: Python/Program/Motherboard.py
from Komponen import Komponen

class Motherboard(Komponen):
    def __init__(self, merk, nama, socket, chipset, jumlahSlotRam):
        super().__init__(merk, nama)
        self.socket = socket
        self.chipset = chipset
        self.jumlahSlotRam = jumlahSlotRam
        
    def set_socket(self, socket):
        self._socket = socket
        
    def get_socket(self):
        return self._socket
    
    def set_chipset(self, chipset):
        self._chipset = chipset
        
    def get_chipset(self):
        return self._chipset
    
    def set_jumlahSlotRam(self, jumlahSlotRam):
        self._jumlahSlotRam = jumlahSlotRam
        
    def get_jumlahSlotRam(self):
        return self._jumlahSlotRam

File: Java/Program/Motherboard.java
public class Motherboard extends Komponen 
{

    private String socket;
    private String chipset;
    private int jumlahSlotRam;

    public Motherboard(String merk, String nama, String socket, String chipset, int jumlahSlotRam) 
    {
        super(merk, nama);
        this.socket = socket;
        this.chipset = chipset;
        this.jumlahSlotRam = jumlahSlotRam;
    }

    public void setSocket(String socket) 
    {
        this.socket = socket;
    }

    public String getSocket() 
    {
        return socket;
    }

    public void setChipset(String chipset) 
    {
        this.chipset = chipset;
    }

    public String getChipset() 
    {
        return chipset;
    }

    public void setJumlahSlotRam(int jumlahSlotRam) 
    {
        this.jumlahSlotRam = jumlahSlotRam;
    }

    public int getJumlahSlotRam() 
    {
        return jumlahSlotRam;
    }
}
